cmd/juju/action/cancel.go

```diff
@@ -87,14 +87,15 @@ func (c *cancelCommand) Run(ctx *cmd.Context) error {
 		return errors.Errorf("identifier(s) %q matched action(s) %q, but no actions were canceled", c.requestedIds, actionTags)
 	}
 
-	var unCanceledActions []string
+	var unCanceledActions []names.ActionTag
 	var canceledActions []params.ActionResult
-	for _, result := range actions.Results {
-		if result.Error != nil {
-			unCanceledActions = append(unCanceledActions, result.Action.Tag)
-			continue
+
+	for i, result := range actions.Results {
+		if result.Action != nil {
```

Comment:
The existing code does provide for items 1 and 3. Item 2 is provided in the way of a mongo Assertion that returns a vanilla error with the message "transaction aborted". There may be multiple reasons why the database code, as is, obscures our ability to handle item 2 with a fine degree of accuracy. In essence, right now it is difficult to ascertain why a particular transaction was aborted with certainty.